rust/src/storage/file/rename.rs

```diff
@@ -0,0 +1,109 @@
+use crate::StorageError;
+use std::ffi::CString;
+
+#[cfg(target_os = "linux")]
+const RENAME_NOREPLACE: libc::c_uint = 1;
+
+#[cfg(target_os = "linux")]
+extern "C" {
+    fn renameat2(
+        olddirfd: libc::c_int,
+        oldpath: *const libc::c_char,
+        newdirfd: libc::c_int,
+        newpath: *const libc::c_char,
+        flags: libc::c_uint,
+    ) -> libc::c_int;
```

Comment:
@houqp I figured out linux error, path is required to be absolute, but in tests it is given as relative. 
One way to achive that is to call `fs::canonicalize` within `FileStorageBackend::new` constructor, wdyt?

Reply:
yep, that's a good idea :+1: